lib/session.js

```diff
@@ -46,6 +114,16 @@ Session.prototype._getSshConnInfo = function() {
 };
 
 Session.prototype._withSshClient = function(callback) {
+  var self = this;
+
+  if(this._ssh_config === null) {
```

Comment:
Here's how to do this.

#### Without Keepalive
Easy, call the `_getSshConfigInfo` method everytime. it's possible to change the host file by some process. So, we need to read it all the time.

#### With keepalive
When asked for keepaalive, we need to cache the existing config. So, we only need to pick it once. User knows the consequence of caching the host info.

If we started fetching the process, you may need to queue until if fetches the hostfile.